libraries/aws/provisioning/include/aws_iot_provisioning.h

```diff
@@ -115,12 +116,59 @@ AwsIotProvisioningError_t AwsIotProvisioning_Init( uint32_t mqttTimeout );
  *   #AWS_IOT_PROVISIONING_INTERNAL_FAILURE, if any there are operation failures internal to the library.
  */
 /* @[declare_provisioning_createkeysandcertificate] */
-AwsIotProvisioningError_t AwsIotProvisioning_CreateKeysAndCertificate( IotMqttConnection_t provisioningConnection,
+AwsIotProvisioningError_t AwsIotProvisioning_CreateKeysAndCertificate( IotMqttConnection_t connection,
                                                                        uint32_t flags,
                                                                        uint32_t timeoutMs,
                                                                        const AwsIotProvisioningCreateKeysAndCertificateCallbackInfo_t * pResponseCallback );
 /* @[declare_provisioning_createkeysandcertificate] */
 
+/**
+ * @brief Requests the AWS IoT Core service for a certificate by sending a Certificate-Signing Request, and
+ * invokes the provided user-defined response handler with the response it receives from the server.
+ *
+ * @note It is advised to use a shared MQTT connection to AWS IoT Core across all API functions.
+ *
+ * @warning This function is NOT thread-safe. Concurrent calls to the library API functions can result in undefined
+ * behavior. Device provisioning with this library REQUIRES calling the API functions of this library sequentially.
+ *
+ * @param[in] connection The MQTT connection handle that will be used to communicate with AWS IoT Core for
+ * the Certificate-Signing Request.
+ * @param[in] operationQos The Quality of Service (QoS) level for the MQTT publish/subscribe communication with the server.
+ * @param[in] pCertificateSigningRequest The PEM encoded string for the Certificate-Signing Request.
```

Comment:
I agree, that is something the service-side doesn't currently support. I can initiate a conversation with them to see if they could support it before the GA. 
From the SDK perspective, it would be simple to accommodate both DER and PEM encoding